src/modules/workspaces/guards/workspaces.guard.ts

```diff
@@ -73,8 +81,17 @@ export class WorkspaceGuard implements CanActivate {
     const isUserNotInWorkspace = !workspaceUser;
     const isRequiredOwnerRoleAndUserIsNotOwner =
       role === WorkspaceRole.OWNER && !workspace.isUserOwner(user);
+    const isWorkspaceNotSetup = !workspace.isWorkspaceReady();
+
+    if (isWorkspaceNotSetup && workspace.isUserOwner(user)) {
+      return true;
+    }
 
-    if (isUserNotInWorkspace || isRequiredOwnerRoleAndUserIsNotOwner) {
+    if (
+      isUserNotInWorkspace ||
+      isRequiredOwnerRoleAndUserIsNotOwner ||
+      isWorkspaceNotSetup
```

Comment:
This is a bit misleading. In reality, you are rejecting when the workspace is not set up and the user is not the owner.